### ISSUE TITLE
Does not work on Cygwin

### ISSUE THREAD

**Issue by DancingQuanta** [rhysd/vim-grammarous#30]
The call to execute LanguageTool in Cygwin fails unsurprisely as the Java on Windows is unable to recognise the posix path of the jarfile given to it.
A fix involves passing the posix path through cygpath command then to Java.
I does not have any vimscript experience, only Python so I may not be able to implement a fix any time soon.

**Comment by rhysd** [comment]
Unfortunately, I don't (and won't) use Cygwin.  So I can't reproduce it.  If you describe the detailed behavior you want,  I may be able to implement it.

**Comment by DancingQuanta** [comment]
Sorry for the late reply.
I am using cygwin on windows which means that I am using Windows native Java runtime.
Windows Java only understands Windows path like this `C:/path-to-text-file.txt` while Cygwin use different path scheme like this `cygdrive/c/path-to-file.txt`.
Cygwin have a helpful program called `cygpath` that converts between Cygwin and Windows paths.
Here is my shell script that call Languagetool jarfile that i have been using before i discovered your Grammarous
```
#!/usr/bin/sh
LANGUAGETOOL=$HOME/bin/LanguageTool-3.5/languagetool-commandline.jar
java -jar $(cygpath -aw $LANGUAGETOOL) "$(cygpath -wa $1)"
```
The shell script converts the cygwin path of both LanguageTool jarfile and text file to windows paths that Java recognises.

Here is the error resulting from executing `:GrammarousCheck` after copying LanguageTool files to `misc`
```
Calling shell to execute: "(java -jar /cygdrive/c/Users/andre/.vim/dein/repos/github.com/rhysd/vim-grammarous/misc/languagetool-commandline.jar -c utf-8 -d 'WHITESPACE_RULE,EN_QUOTES' -l en --api /tmp/v3VBiYW/0) >/tmp/v3VBiYW/1 2>&1"
vim-grammarous: Command 'java -jar /cygdrive/c/Users/andre/.vim/dein/repos/github.com/rhysd/vim-grammarous/misc/languagetool-commandline.jar -c utf-8 -d 'WHITESPACE_RULE,EN_QUOTES' -l en --api /tmp/v3VBiYW/0' failed:
Error: Unable to access jarfile /cygdrive/c/Users/andre/.vim/dein/repos/github.com/rhysd/vim-grammarous/misc/languagetool-commandline.jar^M
Error detected while processing function grammarous#check_current_buffer[16]..grammarous#get_errors_from_xml:
line    1:
E730: using List as a String
E116: Invalid arguments for function filter(a:xml.childNodes(), 'v:val.name ==# "error"'), 's:unescape_error(v:val.attr)')
E116: Invalid arguments for function map(filter(a:xml.childNodes(), 'v:val.name ==# "error"'), 's:unescape_error(v:val.attr)')
E15: Invalid expression: map(filter(a:xml.childNodes(), 'v:val.name ==# "error"'), 's:unescape_error(v:val.attr)')
Yay! No grammatical errors detected.
```

**Comment by ingber** [comment]
Unfortunately, I don't (and won't) use Java.  Any chance of having a fork without Java?  I'd like to use LT in Cygwin Vim.

Thanks for the Desktop Chrome Extension and for the Android speller.

**Comment by DancingQuanta** [comment]
LanguageTool is built using Java unfortunately.

**Comment by rhysd** [comment]
@ingber 

As @DancingQuanta said, LanguageTool is built with Java. And this plugin itself doesn't use Java. You should comment it on LanguageTool ML.

**Comment by rhysd** [comment]
@DancingQuanta Sorry for late response. Thank you for detailed information about Cygwin. Based on your comment, I'll try to add converting paths. As previous comment, I don't have cygwin environment. So please check it works after I finish the fix.

**Comment by rhysd** [comment]
@DancingQuanta 

I tried to fix this based on your information at 33f9b3a. Could you try the latest?

**Comment by rhysd** [comment]
This should be fixed in 33f9b3a and 6057588. Thank you for reporting and fix.
